collections_hierarchy/mixins.py

```diff
@@ -1,87 +1,143 @@
-class ComparableMixin(object):
+from copy import deepcopy
+
+class ComparableMixin(object): # Subclassed in List & Dict
     def __eq__(self, other):
-        pass
+        data = getattr(
+            self,
+            self.DATA_ATTR_NAME
+        )
+        data2 = getattr(
+            other,
+            other.DATA_ATTR_NAME
+        )
+        return data == data2
 
     def __ne__(self, other):
         return not self == other
 
     def __lt__(self, other):
-        pass
+        data = getattr(
+            self,
+            self.DATA_ATTR_NAME
+        )
+        data2 = getattr(
+            other,
+            other.DATA_ATTR_NAME
+        )
+        return data < data2
 
     def __gt__(self, other):
-        pass
+        data = getattr(
+            self,
+            self.DATA_ATTR_NAME
+        )
+        data2 = getattr(
+            other,
+            other.DATA_ATTR_NAME
+        )
+        return data > data2
 
     def __le__(self, other):
-        pass
+        return not self > other
 
     def __ge__(self, other):
-        pass
+        return not self < other
 
 
-class SequenceMixin(object):
+class SequenceMixin(object): # Subclassed in List & Dict
     def __iter__(self):
-        pass
+        return iter(self.get_elements())
 
     def __next__(self):
-        pass
+        return next(iter(self))
 
     next = __next__
 
     def __len__(self):
-        pass
+        return sum([1 for item in self])
+    
```

Comment:
I saw another group use a similar solution for a different problem. They made a comprehension without the list brackets and summed it. I tested in iPython and it works without them.